full-node/sov-stf-runner/src/runner.rs

```diff
@@ -201,6 +201,19 @@ where
                     .prove(header_hash)
                     .await
                     .expect("The proof creation should succeed");
+
```

Comment:
Since the `run_in_process` method is a for loop, does this new proof submission & check loop mean that we block on proof generation before we move on to executing the next block? 

Since we want to provide soft confirmations and generally stay on par with the DA layer natively, we likely don't want this to happen. Are we doing this to decrease PR size for now?

Reply:
Certainly, we're currently implementing this as a temporary measure. The plan is to share the proof only after the relevant block has been finalized. However,  we haven't implemented the logic for `block-finalization` yet. Once Nikolai completes the `fork-management` feature, we'll integrate this logic in the appropriate location.